renovate/src/Renovate/BasicBlock/Assemble.hs

```diff
@@ -113,21 +119,30 @@ assembleBlocks mem isa absStartAddr absEndAddr origTextBytes extraAddr assemble
                        , asExtraAddr        = extraAddr
                        , asExtraText        = mempty
                        , asAssemble         = assemble
-                       , _asOrigBlocks      = L.sortOn basicBlockAddress filteredBlocks
-                       , _asAllocatedBlocks = L.sortOn basicBlockAddress allocatedBlocks
+                       , _asOrigChunks      = L.sortOn chunkAddress filteredBlocks
+                       , _asAllocatedChunks = L.sortOn chunkAddress allocatedBlocks
                        , asISA              = isa
                        , asMemory           = mem
                        , asInstructionConstraints = id
                        }
     -- Split the inputs block list into 2 lists. One for blocks that fit in the
     -- original address space and one for the newly allocated blocks.
-    (origBlocks, allocatedBlocks) = foldr go ([],[]) blocks
+    --
+    -- We mix basic blocks and raw injected code using the 'Chunk' abstraction
+    -- below.  Most of the code here doesn't really need to know anything
+    -- besides the address and size of each chunk (and how to turn a chunk into
+    -- a bytestring).
+    allCode = map BlockChunk blocks ++ [ RawChunk addr bytes | (addr, bytes) <- injectedCode ]
```

Comment:
How does the raw injected code in a `Chunk` interact with transforms in SFE? For example, if an earlier transform introduces raw injected code, do later transforms see that raw code? Can they do anything with it (e.g. disassemble it)?

Reply:
The code injection API is not intended to be used with SFE.  If we find a use for it in SFE, I'd say that we probably won't want SFE to see any of the injected code.  I can see a potential future where we'll want the ability to inject non-raw code (i.e., as a list of instructions instead of a bytestring).  I'm willing to put off those decisions until a use case comes up.